core-js/src/main/javascript/cdf/dashboard/Dashboard.lifecycle.js

```diff
@@ -638,6 +638,9 @@ define([
       }
 
       if(object.isManaged === false && object.update) {
+        if (object.defaultValue == null && !object.useFirstValue)  {
```

Comment:
It might be better to not add component logic into the Dashboard sub-modules.

Reply:
@afrjorge why not ? this logic needed in dashboard

Reply:
But it's not dashboard behavior, it's component behavior. `object` is the component, and only 5 components use the `defaultIfEmpty` (`useFirstValue`), so it's not even general component behavior, it's specific to components that extend from the SelectBase or ToggleButtonBase components.

Reply:
@afrjorge I've moved logic into components, please look at it

Reply:
When a user creates a CDE dashboard he will not be able to, for example, add a select component and have the first value selected because, in this case, _defaultIfEmpty && useFirstValue_ (defaults _true && undefined_) is _falsy_, and because the properties are not editable in the CDE editor, selecting the first value will not be possible.